notebooks/scrubbed/covid-19-tests-august-2020.ipynb

```diff
@@ -0,0 +1,599 @@
+    "- *wisdom of the crowds*: incorporate other predictions, namely the [Metaculus one](https://pandemic.metaculus.com/questions/4400/by-august-1-how-many-tests-for-covid-19-will-have-been-administered-in-the-us/), so we can include information we might have missed.\n",
+    "\n",
+    "## Question decomposition\n",
+    "\n",
+    "Let's decompose our question into a number of sub questions. We'll then focus on trying to estimate each of these separately, allowing us to iteratively improve our model as we go along.\n",
+    "\n",
+    "So the number of tests done by August 1st could be decomposed as:\n",
+    " - ensemble prediction of:\n",
+    "     - my prediction\n",
+    "        - current number of tests: how many tests did we do today?\n",
+    "        - how much we can expect to change until August 1st?\n",
+    "            - if we repeat our current rates, how does it look in August?\n",
+    "            - how much we can we expect our testing rates to change based on the past results?\n",
+    "            - how much can we expect our testing rates to change, based on future factors?\n",
+    "     - metaculus question "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# My Prediction\n",
+    "\n",
+    "## Get relevant data\n",
+    "\n",
+    "To know the current testing rate, and and how it changes, we're going to use the https://covidtracking.com/ API to fetch the testing statistics. To make it easier to parse, we're going to filter to the fields we care about:\n",
+    "- date\n",
+    "- totalTestResults: culmulative tests per day\n",
+    "- totalTestReusltsIncrease: how many new test results were reported that day"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "testing_data = pd.read_csv(\"https://covidtracking.com/api/v1/us/daily.csv\")\n",
+    "# testing data uses numbers for dates (20200531 for e.g.), so let's convert that to use python dates instead\n",
+    "testing_data[\"date\"] = testing_data[\"date\"].apply(lambda d: datetime.strptime(str(d),  \"%Y%m%d\"))\n",
+    "\n",
+    "# now filter just to the columns we care about\n",
+    "testing_data = testing_data[[\"date\", \"totalTestResults\", \"totalTestResultsIncrease\"]]\n",
+    "testing_data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Current number of tests\n",
+    "\n",
+    "We can now use our dataset to easily find the number of tests done today (or the latest in the dataset), which we can verify is correct by looking at the table above"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# pick the largest, therefore most up to date, date\n",
```

Comment:
That's fair - I'm making a weaker assumption here, since their CSV API doesn't seem to have an ordering guarantee, so don't want this to break if they change that.